crates/settlement-clients/starknet/src/tests/test.rs

```diff
@@ -0,0 +1,229 @@
+use crate::StarknetSettlementClient;
+use rstest::{fixture, rstest};
+use settlement_client_interface::SettlementClient;
+use starknet::{
+    accounts::{Account, ConnectedAccount, ExecutionEncoding, SingleOwnerAccount},
+    contract::ContractFactory,
+    core::types::{
+        contract::{CompiledClass, SierraClass},
+        BlockId, BlockTag, DeclareTransactionResult, Felt, FunctionCall, InvokeTransactionResult, StarknetError,
+        TransactionExecutionStatus, TransactionStatus,
+    },
+    macros::{felt, selector},
+    providers::{jsonrpc::HttpTransport, JsonRpcClient, Provider, ProviderError, Url},
+    signers::{LocalWallet, SigningKey},
+};
+use std::env;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Duration;
+use utils::settings::env::EnvSettingsProvider;
+use utils::settings::Settings;
+
+pub async fn spin_up_madara() -> MadaraCmd {
+    println!("Spinning up madara");
+    env::set_current_dir(env::var("MADARA_BINARY_PATH").unwrap()).expect("Failed to set working directory");
+    println!("Current working directory: {:?}", env::current_dir().unwrap());
+    let mut node = MadaraCmdBuilder::new()
+        .args([
+            "--network",
+            "devnet",
+            "--no-sync-polling",
+            "--authority",
+            "--devnet",
+            "--preset=test",
+            "--no-l1-sync",
+            "--rpc-cors",
+            "all",
+            "--chain-config-override",
+            "--block-time",
+            "1",
+        ])
+        .run();
+    node.wait_for_ready().await;
+    node
+}
+
+async fn wait_for_tx_success(
+    account: &SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>,
+    transaction_hash: Felt,
+    duration: Duration,
+) -> bool {
+    let mut attempt = 0;
+    loop {
+        attempt += 1;
+        if attempt >= 5 {
+            return false;
+        }
+        let reciept = match account.provider().get_transaction_status(transaction_hash).await {
+            Ok(reciept) => reciept,
+            Err(ProviderError::StarknetError(StarknetError::TransactionHashNotFound)) => {
+                tokio::time::sleep(duration).await;
+                continue;
+            }
+            _ => panic!("Unknown error"),
+        };
+
+        match reciept {
+            TransactionStatus::Received => (),
+            TransactionStatus::Rejected => return false,
+            TransactionStatus::AcceptedOnL2(status) => match status {
+                TransactionExecutionStatus::Succeeded => return true,
+                TransactionExecutionStatus::Reverted => return false,
+            },
+            TransactionStatus::AcceptedOnL1(status) => match status {
+                TransactionExecutionStatus::Succeeded => return true,
+                TransactionExecutionStatus::Reverted => return false,
+            },
+        }
+        // This is done, since currently madara does not increment nonce for pending transactions
+        tokio::time::sleep(duration).await;
```

Comment:
instead of this, should we wait for block to be not pending?

Reply:
I assume, this achieves the same, also I am not sure how does one go from transaction hash to block status